src/all_implementations.jl

```diff
@@ -307,7 +307,7 @@ Base.length(layout::EmptyArray) = 0
 Base.firstindex(layout::EmptyArray) = 1
 Base.lastindex(layout::EmptyArray) = 0
 
-Base.eltype(layout::EmptyArray) = nothing
+Base.eltype(layout::EmptyArray) = Union{}
```

Comment:
Yes, that's right. You don't get "nothing" out of an `EmptyArray`, you get the bottom type ("impossible"). Also, `nothing` (lowercase "n") is the singleton instance, not the type (which has a capital "N").

Reply:
indeed, here's a fun piece:

```julia
julia> typeof(Union{})
Core.TypeofBottom

julia> Base.Bottom
Union{}
```